REGRESSION (iOS 15): accessibility/misspelling-range.html is constantly failing

AndresGonzalezApple opened the pull request:
#### a7cbc19642ca8032fcec544d951b328cdc910eaa
<pre>
REGRESSION (iOS 15): accessibility/misspelling-range.html is constantly failing
<a href="https://bugs.webkit.org/show_bug.cgi?id=231261">https://bugs.webkit.org/show_bug.cgi?id=231261</a>
<a href="https://rdar.apple.com/80383672">rdar://80383672</a>

Reviewed by Tyler Wilcock.

This test was failing because it needs to be async for the unified checker to work.
Re-wrote the test to match LayoutTests/accessibility standards.

* LayoutTests/accessibility/misspelling-range-expected.txt:
* LayoutTests/accessibility/misspelling-range.html:
* LayoutTests/platform/ios/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/282537@main">https://commits.webkit.org/282537@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b2a6fec50ccf07d22e1a66fee3edae5f35fb0c5c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/63330 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/42686 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/15927 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/67351 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/13938 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/50374 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/14218 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/51021 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/9639 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/66399 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/39629 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/54844 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/31702 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/36312 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/12194 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/12810 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/57851 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/12521 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/69047 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/7277 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/12123 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/58334 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/7308 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/54915 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/58580 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14050 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/6067 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/38507 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/39586 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/40698 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/39329 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

hoffmanjoshua commented on LayoutTests/accessibility/misspelling-range.html:
```diff
@@ -1,89 +1,90 @@
-        var content = document.getElementById("content");
-        content.focus();
+let output = "This tests that misspelling ranges are properly retrieved in the fashion that a spell checker would.\n\n";
+jsTestIsAsync = true;
```
nit: `jsTestIsAsync` should probably go inside the `if (window.accessibilityController && window.internals)` condition, in case that fails.

AndresGonzalezApple committed:
REGRESSION (iOS 15): accessibility/misspelling-range.html is constantly failing
https://bugs.webkit.org/show_bug.cgi?id=231261
rdar://80383672

Reviewed by Tyler Wilcock.

This test was failing because it needs to be async for the unified checker to work.
Re-wrote the test to match LayoutTests/accessibility standards.

* LayoutTests/accessibility/misspelling-range-expected.txt:
* LayoutTests/accessibility/misspelling-range.html:
* LayoutTests/platform/ios/TestExpectations:

Canonical link: https://commits.webkit.org/282537@main
